### DISCUSSION TITLE
tool to extract hardcoded subs in videos

### DISCUSSION THREAD

**Discussion by eadmaster** [rakuri255/UltraSinger#132] [Q&A]
which tool do you suggest to extract subs with sub-word precision  from yt videos?

All the tools i've found seems to be static images/line-oriented, and thus are not adeguate for karaoke use:

- https://github.com/YaoFANGUK/video-subtitle-extractor
- https://github.com/devmaxxing/videocr-PaddleOCR
- https://github.com/IvRogoz/pyDetectSub
- https://sourceforge.net/projects/videosubfinder/

**Comment by rakuri255** [comment]
Its a good idea to use the subs from youtube!

We should first look if there is already subs form the yt downloader. 
When there are only burn-in subs than your tools are a good way.
But we need word timestamps from them!

**Reply by eadmaster** [reply]
yep, i am looking for a tool that can ocr from burn-in subs, with word timestamps.

It has to detect the highlighted part of each line.

Some example videos:

 - https://www.youtube.com/watch?v=Wur9gMOOur4
 - https://www.youtube.com/watch?v=jd7fdiTWZQU
